host-bin/edit-chroot

```diff
@@ -292,8 +317,11 @@ for NAME in "$@"; do
             error 2 "$CHROOT already exists! Specify a second -r to overwrite it (dangerous)."
         elif [ -n "$RESTORE" ]; then
             EXISTS='y'
-        else
-            sh -e "$BINDIR/unmount-chroot" $YESPARAM -c "$CHROOTS" -- "$NAME"
+        elif ! sh -e "$BINDIR/unmount-chroot" $YESPARAM \
+                    -c "$CHROOTS" -- "$NAME"; then
+            if [ -n "$NEEDS_UNMOUNT" ]; then
+                exit 1
+            fi
```

Comment:
Because we manually mounted with mount-chroot, so we'd like to put it back the way we found it.  If the chroot is active, unmount-chroot will fail, which is OK if it doesn't really need to be unmounted to continue.